### DISCUSSION TITLE
Is it possible to add the functionality to specify tags in the @BeforeSuite and @AfterSuite hooks.

### DISCUSSION THREAD

**Discussion by zbrea** [getgauge/gauge#2110] [General]
> I got addicted to gauge guys! (even on weekends) :)

I would like to know what prevents adding the functionality to specify tags in the @BeforeSuite and @AfterSuite hooks.
In my tests I really need this functionality.

**Comment by sriv** [comment]
The suite hooks run once before and after the whole suite run. So it does not matter what tags are in play, since tags only affect specs/scenarios. Does that make sense?

Did you have a particular scenario in mind for the suite hooks to be accomodating tags?

**Comment by sriv** [comment]
Btw, glad you are linking gauge, thank you for the kind words!

**Comment by zbrea** [comment]
public class DriverFactory {

//Case 1: Test without restarting the application during the entire test suite
//execution command: mvn gauge:execute -DspecsDir=specs -Dtags=**"without_restart_app"**
    
>	@BeforeSuite(tags = {"without_restart_app"})
    public void startServerAndApp() throws Exception {
        startAppiumServer();
        startApp();
    }

>    @AfterScenario(tags = {"without_restart_app"})
    public void reStartApp(ExecutionContext ctx) throws Exception {
        String CurrentScenario = ctx.getCurrentScenario().getName();
        if (ctx.getCurrentScenario().getIsFailing()) {
            startApp();
        }
    }
   .
   .
    
//Case 2: Test with restart of the application for each scenario:
//execution command: mvn gauge:execute -DspecsDir=specs -Dtags=**"with_restart_app"**

>   @BeforeSuite(tags = {"with_restart_app"})
    public void startServer() throws Exception {
        startAppiumServer();
    }

>  @BeforeScenario(tags = {"with_restart_app"}) 
    public void startAppDriver() throws Exception {
            startApp();
    }
    .
    .

//Case 3: Test with with_serveur SauceLabs:
//execution command: mvn gauge:execute -DspecsDir=specs -Dtags=**"with_SauceLabs"**

  >   @BeforeSuite(tags = {"with_SauceLabs"})
    public void startSauceLabs() throws Exception {
        startSauceLabsDriver();
    }
   .
   .

}

**Reply by sriv** [reply]
I believe this is what you can use the environment for. i.e. have environments defined for each such cases and these can contain property files that indicate whether you need a restart, or if you should use sauselab. This will then let you run `mvn gauge:execute =Denv=sauselabs"

**Reply by zbrea** [reply]
Thanks for this info, I never used this environment feature, I will see how to use it.
If I find some examples on github it will be very cool

**Reply by sriv** [reply]
Please do read this section in the docs - https://docs.gauge.org/configuration.html#using-environments-in-a-gauge-project

For examples - please see https://github.com/getgauge-examples/java-maven-selenium on how you can use environments to run the same tests on different browsers.

**Reply by zbrea** [reply]
Thank you for your indications !
